crates/nu-cli/src/completions/file_completions.rs

```diff
@@ -157,3 +151,55 @@ pub fn matches(partial: &str, from: &str) -> bool {
     from.to_ascii_lowercase()
         .starts_with(&partial.to_ascii_lowercase())
 }
+
+// escape paths that contains some special characters
+pub fn escape_path_str(path: String) -> String {
+    let mut path = path;
+
+    // List of special characters that need to be escaped
+    let special_characters = b"\\\'\"";
+    let replacements = [b"\\\\", b"\\\'", b"\\\""];
+
+    // Check if path needs to be escaped
+    let needs_escape = path.bytes().fold(false, |acc, x| {
+        acc
+        || x == b'\\' // 0x5c
+        || x == b'`' // 0x60
+        || x == b' '
+        || x == b'\''
+    });
+
+    if needs_escape {
+        // Worst case, escape everything (+2 double quotes)
+        let mut result = String::with_capacity(path.len() * 2 + 2);
+
+        // Starting quotes
+        result.push('\"');
+
+        // Walk through the path characters
+        for b in path.bytes() {
+            // Basically the equivalent of str.find(), but expanded
+            if let Some(idx) = special_characters.iter().enumerate().fold(None, |idx, c| {
+                if *c.1 == b {
+                    Some(c.0)
+                } else {
+                    idx
+                }
+            }) {
+                for rb in replacements[idx] {
+                    result.push(*rb as char);
```

Comment:
This is wrong, since not all bytes we iterate through are valid `char`s on their own (some might be part of a multibyte character). See the updated code in my earlier comment.

Reply:
The way you wrote doesn't work

Reply:
```
mismatched types
expected `char`, found `&u8`
```

Reply:
Are you looking at the *updated* version of the comment?